lib/elixir/lib/task.ex

```diff
@@ -344,7 +344,8 @@ defmodule Task do
   This function returns a tuple with the returned value
   in case the message matches a task that exited with
   success alongside the matching task. It returns `nil`
-  if no task was found. It exits if the task has failed.
+  if no task was found. It exits if any of the tasks have
+  failed.
```

Comment:
This change is misleading. The return or exit is based purely on the message and so relates to "the" task. Another task could have failed but it won't cause an exit.